src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,721 @@
+        e.preventDefault();
+        let nextHighlightedValue = highlighted;
+        const highlightedIndex = navigatable.current.findIndex(
+          value => value === highlighted
+        );
+        if (e.key === "ArrowUp") {
+          if (highlightedIndex === 0) {
+            nextHighlightedValue = navigatable.current[navigatable.current.length - 1];
+          } else {
+            nextHighlightedValue = navigatable.current[highlightedIndex - 1];
+          }
+        } else if (e.key === "ArrowDown") {
+          e.preventDefault();
+          if (highlightedIndex === navigatable.current.length - 1) {
+            nextHighlightedValue = navigatable.current[0];
+          } else {
+            nextHighlightedValue = navigatable.current[highlightedIndex + 1];
+          }
+        } else if (e.key === "End") {
+          e.preventDefault();
+          nextHighlightedValue = navigatable.current[navigatable.current.length - 1];
+        } else if (e.key === "Home") {
+          nextHighlightedValue = navigatable.current[0];
+          e.preventDefault();
+        }
+        setHighlighted(nextHighlightedValue);
+      }
+    }
+  };
+  const isHidden = useCallback(
+    (value?: string) => {
+      return !visibleList.current.includes(value ?? "");
+    },
+    [visibleList]
+  );
+
+  const optionContextValue = useMemo(() => {
+    return {
+      search,
+      updateHighlighted: setHighlighted,
+      highlighted,
+      isHidden,
+      onSelect,
+      selectedValue: value,
+    };
+  }, [search, highlighted, isHidden, onSelect, value]);
+
+  const conditionalProps: Partial<SelectOptionProp> = {};
+  if (options) {
+    conditionalProps.options = options;
+  } else {
+    conditionalProps.children = children;
+  }
+
+  return (
+    <FormRoot
+      $orientation={orientation}
+      $dir={dir}
+      $addLabelPadding
+      {...props}
+    >
+      <FormElementContainer>
+        <SelectPopoverRoot
+          open={open}
+          onOpenChange={onOpenChange}
+        >
+          <StyledSelectTrigger
+            id={id ?? defaultId}
+            $error={!!error}
+            disabled={disabled}
+            data-testid="autocomplete-trigger"
+          >
+            <Icon
+              name="search"
+              size="sm"
+            />
+            <Placeholder>{placeholder}</Placeholder>
+          </StyledSelectTrigger>
+          <Portal>
+            <PopoverContent
+              sideOffset={0}
+              onFocus={onFocus}
+              onCloseAutoFocus={() => {
+                onUpdateSearch("");
+              }}
+              onOpenAutoFocus={() => {
+                setHighlighted(visibleList.current[0]);
+              }}
+              align="start"
+            >
+              <SearchField
+                ref={inputRef}
+                value={search}
+                onChange={onUpdateSearch}
+                onKeyDown={onKeyDown}
+                placeholder={placeholder}
+              />
```

Comment:
we have our Popover component, why are we recreating another one here? with styles, key actions etc? It should be a functionality of our Popover component, we shouldn't have this code duplication

Reply:
Our popover has different styling and also we need the width of the element so I thought it would be better to create it separately
Also in case we are changing something in popover it should not affect here